scripts/ci/libraries/_build_images.sh

```diff
@@ -112,27 +112,60 @@ function build_images::forget_last_answer() {
     fi
 }
 
-function build_images::confirm_via_terminal() {
-    echo >"${DETECTED_TERMINAL}"
-    echo >"${DETECTED_TERMINAL}"
+
+function build_images::reconfirm_rebuilding_if_not_rebased() {
```

Comment:
Rather than handling all the different cases when user is not rebased, we now simply (by default) check and require the branch of the user to be rebased to latest `main` when rebuilding the image (and warn the user that not doing so might take a long time). This cuts down the number of cases that we had to handle and eliminates the need of storing python base image in ghcr.io (because different branches and  locally built image  could have different base image).

By asking the user to rebase when rebuilding we make sure that the same base python image is used (and also it makes it possible to use `--pull` flag on all rebuilds to always pull the latest base python image. 

This might have a side effect only for a very short while when the new image is relased and no "main" build succeeded yet - but with the recent stability improvements we had in flaky tests, this case should be very limited.